### DISCUSSION TITLE
Oxygen builder integration

### DISCUSSION THREAD

**Discussion by mg33chicago** [swup/swup#892] [Help]
### Describe your problem 🧐

I have Swup running in Oxygen Builder in Wordpress. It's working great, save for one problem: on mobile, sometimes clicking a navigation link to a new page results in a page that can't be scrolled (content is cut off at the bottom), and, sometimes the styles don't fully load for the homepage. It's random, and I can't figure out why. Sometimes the Projects page is affected by this, sometimes the Information page, and sometimes the Get In Touch page. 

### What is your question? 🙋‍♀️

How can I troubleshoot, and resolve this issue?  Is there an additional feature I need to enable to ensure that new pages load 100% and all stylesheets that affect the page fully load? Am I even looking at this problem the right way, or is there something I'm missing? To reproduce, the browser inspector and mobile view is encountering the same problem as a mobile device. The homepage not fully loading occurs on desktop and on mobile, but it's random. I can't quite figure out why it happens sometimes and not others. 

Also worth noting: all of these issues resolve with a browser refresh. They do reoccur, but when they're happening, a browser refresh clears it up temporarily. 

### URL to minimal reproduction 🔗

https://mguxd.com/home-2024

### Checked all these? 📚

- [X] I have read and searched the [official docs](https://swup.js.org)
- [X] I have checked [discussions](https://github.com/orgs/swup/discussions) and [existing issues](https://github.com/issues?q=is:issue+org:swup) for answers to my question
- [X] I agree to follow this project's [Code of Conduct](https://github.com/swup/.github/blob/master/CODE_OF_CONDUCT.md)
- [X] I have provided all necessary information to the best of my knowledge

**Comment by daun** [comment]
On first glance, it looks like you're initialising swup twice.

```html
<script src="https://unpkg.com/swup@4"></script>
<script>
  const swup = new Swup();
</script>
<script src="https://unpkg.com/@swup/scripts-plugin@2"></script>
<script>
  const swup = new Swup({
    plugins: [new SwupScriptsPlugin()]
  });
</script>
```

**Comment by mg33chicago** [comment]
Thank you. I've cleaned that up and trying to test it, however, because I pulled out every but of JS I was using for a few things, I'm back to my problem of the expanded mobile menu not closing when a link inside is clicked. It remains open. 

The class for the expanded state is .oxy-nav-menu.oxy-nav-menu-open, and the closed state it needs to return to upon click is .oxy-nav-menu.oxy-nav-menu-closed.

Pages are reloading behind the menu because I can see the url change. But the menu remains visible. Is this something that Swup needs to trigger the closure of? I previously had some custom JS in there that was working, but trying to keep things as clean as possible and in the confines of what Swup should do to handle page transitions.

Update: I added in some code to close the menu when viewport is below 992px in width. Swup seems to be installed correctly and the page transitions are happening, but it is still doing the thing where it sometimes only loads a partial page, and the page can't be scrolled. 

Thanks again, I really appreciate your help.

**Reply by daun** [reply]
Swup only handles the loading of and transition to new pages. Other client-side logic, like closing the menu as mentioned, is something you‘ll need to do yourself. Adding a hook whenever a visit starts should work in that case.

```js
swup.hooks.on('visit:start', () => {
  /* close menu */
})

**Reply by mg33chicago** [reply]
Thanks, I so appreciate your help. I was ecstatic to find Swup last weekend after toiling for hours to get a smooth page transition on this site. I'm so close... I'm going to need to keep working on the JS to close the menu.

I'm using Oxygen Builder, and my Swup ID is only on an inner container in my master page template. If there's a way to more selectively determine why these pages aren't loading, I'd love a suggestion. I'm quite familiar with inspectors and just can't figure out how to tel what the problem is. I'll add that when I took Swup out, the pages worked fine.

**Reply by mg33chicago** [reply]
I discovered something - FINALLY - after trying a ton of different things. When the problem occurs and the page doesn't scroll properly, the following class has been applied to the opening HTML tag:

<html class="class="swup-enabled oxy-nav-menu-prevent-overflow" lang="en-US">

When the newly loaded page does scroll correctly, the applied class is:
<html class="class="swup-enabled" lang="en-US">

Why might oxy-nav-menu-prevent-overflow sometimes be added as a class?

**Reply by daun** [reply]
I'm not familiar with Oxygen Builder, but if they have a support forum or discord channel, you'll probably get help much more quickly by posting there. This seems to be a problem with how it behaves in combination with ajax-loaded pages, as is common with tools like swup, htmx, turbo, etc.
